rules/windows/process_creation/proc_creation_win_creation_mavinject_process_injection.yml

```diff
@@ -1,40 +1,45 @@
 title: Mavinject Inject DLL Into Running Process
 id: 4f73421b-5a0b-4bbf-a892-5a7fb99bea66
+related:
+    - id: 17eb8e57-9983-420d-ad8a-2c4976c22eb8
+      type: obsoletes
 status: experimental
-author: frack113
+author: frack113, Florian Roth
 date: 2021/07/12
-modified: 2022/07/11
-description: Injects arbitrary DLL into running process specified by process ID. Requires Windows 10.
+modified: 2022/09/07
+description: Detects process injection using the signed Windows tool "Mavinject" via the "INJECTRUNNING" flag or a renamed execution of the tool
 references:
     - https://github.com/redcanaryco/atomic-red-team/blob/f339e7da7d05f6057fdfcdd3742bfcf365fee2a9/atomics/T1218/T1218.md
     - https://github.com/redcanaryco/atomic-red-team/blob/f339e7da7d05f6057fdfcdd3742bfcf365fee2a9/atomics/T1056.004/T1056.004.md
     - https://posts.specterops.io/mavinject-exe-functionality-deconstructed-c29ab2cf5c0e
+    - https://twitter.com/gN3mes1s/status/941315826107510784
+    - https://reaqta.com/2017/12/mavinject-microsoft-injector/
+    - https://twitter.com/Hexacorn/status/776122138063409152  # Deleted tweet
 tags:
     - attack.defense_evasion
-    - attack.collection
+    - attack.privilege_escalation
+    - attack.t1055.001
     - attack.t1218.013
-    - attack.t1056.004
 logsource:
     category: process_creation
     product: windows
 detection:
-    selection_cli:
-        CommandLine|contains|all:
-            - ' /INJECTRUNNING'
-            - '.dll' # space some time in the end
-    selection_img:
-        - OriginalFileName:
+    selection_flag:
+        CommandLine|contains: ' /INJECTRUNNING '
+    selection_renamed:
+        OriginalFileName:
             - 'mavinject32.exe'
             - 'mavinject64.exe'
-        - Image|endswith:  # Event 4688 doesn't have the OriginalFileName field
+    filter_renamed:
+        Image|endswith:
             - '\mavinject32.exe'
             - '\mavinject64.exe'
-    condition: all of selection_*
+    condition: selection_flag or (selection_renamed and not filter_renamed)
```

Comment:
the logic was if you don't have sysmon ,  You can hope the the image is still mavinjectxx.exe in EvenID 4688.

So now the logic is :
- In any case catch `INJECTRUNNING` args
- If you have sysmon (or ETW) check if rename version is started

Right ?

Reply:
Yes. Because the flag "INJECTRUNNING " is unique and required so we don't care about the image name and we have sysmon logs we can also fire on usage of a renamed version.